scenarios.md

```diff
@@ -0,0 +1,225 @@
+
+1. ACME Rockets acquires network monitoring software, copying it to their firewall protected production environment.
+1. As part of the artifact copy, they will copy/proxy the signature validation to trusted resources within their network protected environment.
+
+**Implications of this requirement:**
+
+- In this scenario, the wabbit-networks signature must be validated within the ACME Rockets network. How this is done is open for design. However, the requirement states the signature must be validated without external access. When the artifact is copied to the private/network restricted registry, the signature may need to be copied, and is assumed to be trusted if available in the trusted server within the private network. How ACME Rockets would copy/proxy the signatures is part of the design and UX for a secure, but usable pattern.
+
+### Scenario #6: Multiple Signatures
+
+Customers may require multiple signatures for the following scenarios:
+
+- Validate the artifact is the same as what the vendor provided.
+- Secondarily sign the artifact by the consuming company, attesting to its validity within their production environment.
+- Signatures represent validations through different dev, staging, production environments.
+- Dev environments support any signature, while integration and production environments require mycompany-prod signatures.
+
+#### Scenario 6.1: Dev and Prod Keys
+
+1. A CI/CD solution builds, signs, pushes and deploys a collection of artifacts to a staging environment.
+1. Once integrations tests are completed, the artifacts are signed with a production signature, copying them to a production registry or production set of repositories.
+1. The integration and production orchestrators validate the artifacts are signed with production keys.
+
+#### Scenario 6.2: Approved Vendor/Project Artifacts
+
+A deployment requires a mydb image. The mydb image is routinely updated for security vulnerabilities. ACME Rockets references stable version tags (`mydb:1.0`), assuring they get newly patched builds, but they must verify each new version to be compatible with their environment.
+
+1. The `mydb:1.0` image is acquired from a public registry, imported into private integration registries.
+1. Functional testing is run in the integration environment, verifying the patched `mydb:1.0` image is compatible.
+1. The `mydb:1.0` is tagged with a unique id `mydb:1.0-202002131000` and signed with an ACME Rockets production key.
+1. The retagged image, with both the mydb and ACME Rockets signatures are copied to a prod registry/repository.
+1. The release management system deploys the new `mydb:1.0-202002131000` image.
+1. The production orchestrator validates it's signed with the Acme Rockets production key.
+
+**Implications of this requirement:**
+
+- Multiple signatures, including signatures from multiple sources can be associated with a specific artifact.
+- Original signatures are maintained, even if the artifact is re-tagged.
+
```

Comment:
```suggestion

#### Scenario 7: A repository compromise occurs

An attacker manages to compromise a repository and gain access to all keys on the repository.  This includes access to sign artifacts using keys sorted on the repository's HSM.  

**Implications of this requirement:**

1. The potential damage/risk to users in this case must be limited
2. There must be a secure way for users to recover to a known, secure state and verify this has occurred even in the face of an attacker that can act as a man-in-the-middle on the network.

```

I'll add more of these later, but wanted feedback here first.

Reply:
> This includes access to sign artifacts using keys sorted on the repository's HSM.

I think this is the first scenario talking about the registry having an HSM or stored keys.

Reply:
That's a great point.  My text should indicate that it provides access to keys stored in hardware if they exist.  I certainly don't mean to imply there must (or even should) be a HSM.  I more meant that it should not be assumed to adequately protect a key in this case.

Reply:
I really like the scenario, and it builds on the threat requirement section. If you can reword it to avoid referencing a specific implementation, such as HSM which doesn't work well in cloud environments, we can merge this in as well. It's not to say we wouldn't support HSMs if the customers environment can, rather it's a specific implementation that I'm trying to avoid.

I have been assuming the private signing keys are NOT part of Notary v2, rather externally stored. 
1. so a registry compromise doesn't compromise all signed objects
1. as cloud providers and vendors each have their own, integrated secure key management system. 

There's a separate requirement, from a requirements pr, that talks to Notary being able to integrate with, through some sort of extension model, a cloud providers solution. For a reference implementation, we could use https://github.com/hashicorp/vault.

I'm happy to copy/paste the above text, and modify. Or, commit a slightly modified suggestion.